roles/common/tasks/main.yml

```diff
@@ -3,24 +3,38 @@
   fail: msg='This role only supports Yum or Apt package managers at the moment.  Please do a manual install instead.'
   when: ansible_pkg_mgr != 'yum' and ansible_pkg_mgr != 'apt'
 
-- name: yum install Java 1.7 and some basic dependencies
+- name: Fetch Java version
+  shell: java -version 2>&1 | grep version | awk '{print $3}' | sed 's/"//g' | grep '^.\..' -o || echo 0.0
```

Comment:
FWIW, `grep version | awk '{print $3}'` is the same as `awk '/version/ {print $3}'`. However, you don't need to combine awk, sed, and grep for this. You can simplify this to one grep to get the line, and one grep to extract a number from that line.

`shell: java -version 2>&1 | grep version | grep -o '[0-9][0-9]*\.[0-9][0-9]*' || echo 0`

Alternatively, if you're confident that the first line with a number has the number you want:

`shell: java -version 2>&1 | grep -o -m1 '[0-9][0-9]*\.[0-9][0-9]*' || 0.0`